src/Symfony/Component/Validator/Constraints/BicValidator.php

```diff
@@ -95,5 +106,39 @@ public function validate($value, Constraint $constraint)
 
             return;
         }
+
+        // check against an IBAN
+        $iban = $constraint->iban;
+        $path = $constraint->ibanPropertyPath;
+        if ($path && null !== $object = $this->context->getObject()) {
+            try {
+                $iban = $this->getPropertyAccessor()->getValue($object, $path);
+            } catch (NoSuchPropertyException $e) {
+                throw new ConstraintDefinitionException(sprintf('Invalid property path "%s" provided to "%s" constraint: %s', $path, \get_class($constraint), $e->getMessage()), 0, $e);
+            }
+        }
+        if (!$iban) {
+            return;
+        }
+        $ibanCountryCode = substr($iban, 0, 2);
+        if (ctype_alpha($ibanCountryCode) && substr($canonicalize, 4, 2) !== $ibanCountryCode) {
+            $this->context->buildViolation($constraint->ibanMessage)
+                ->setParameter('{{ value }}', $this->formatValue($value))
+                ->setParameter('{{ iban }}', $iban)
+                ->setCode(Bic::INVALID_IBAN_COUNTRY_CODE_ERROR)
+                ->addViolation();
+        }
+    }
+
+    private function getPropertyAccessor(): PropertyAccessor
+    {
+        if (null === $this->propertyAccessor) {
+            if (!class_exists('Symfony\Component\PropertyAccess\PropertyAccess')) {
```

Comment:
we should use the `class` constant here

Reply:
@xabbuh like `PropertyAccess::class` ?
I haven't tried but I think it will throw an exception if the class does not exist, right?

Reply:
yes like that, and using the `class` constant works even when the class is not present: https://3v4l.org/knSo3

Reply:
@xabbuh thanks!